metabase/metabase
# Refactor Archive App

**Problem:** The archive app is out-of-date with our standards (class vs function, javascript vs typescript, HoC vs hooks, ...). In order to make it easier to fix a bug and maintain in the future, I refactored it to closer match to our desired code standards.

**Context:**
- I use this refactoring to help fix https://github.com/metabase/metabase/issues/24018. I am creating a separate PR to make it easier to review the refactor separate from the updates.
- If you are reviewing, this is the first PR that has priority for reviewing: https://github.com/metabase/metabase/pull/33600

---
**Desired Behavior**/**Acceptance Criteria**
- [ ] Class component is now converted to functional component
- [ ] HoC components are converted to hooks (where possible -> `Search.loadList` still remains)
  - redux selectors
  - useListSelect hook
- [ ] Convert to TypeScript
- [ ] Archive App has no altered behaviors
  ![archive-app](https://github.com/metabase/metabase/assets/22608765/9174ffef-8c2f-4047-916b-3eee34ab7032)

_Tests_
- [x] Un-archiving individual items
- [x] Deleting individual items
- [x] Bulk un-archiving
- [x] Bulk deleting
- [x] Undoing actions

---
**How to Verify**

1. Open up the archive
2. Test out list selecting (select, de-select, select-all, deselect-all)
3. Test out unarchiving and deleting (individual and bulk)
4. Verify that it behaves appropriately
